packages/expect/src/matchers.ts

```diff
@@ -835,7 +835,7 @@ const matchers: MatchersObject = {
     const pass =
       typeof expected === 'string'
         ? received.includes(expected)
-        : expected.test(received);
+        : new RegExp(expected).test(received);
```

Comment:
I doubt there will any issues with instantiating a new RegExp object for every call, but alternatively, I could limit that to only RegExps with the [`global`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/global) flag. Let me know what you think!